hardware_interface/src/component_parser.cpp

```diff
@@ -929,6 +929,22 @@ std::vector<InterfaceDescription> parse_state_interface_descriptions(
   return component_state_interface_descriptions;
 }
 
+void parse_state_interface_descriptions(
+  const std::vector<ComponentInfo> & component_info,
+  std::unordered_map<std::string, InterfaceDescription> & state_interfaces_map)
```

Comment:
In order to help with future-proofing this a little bit I am wondering if we could use a template for this function with some traits or concepts (or none if none works yet w this C++ version) to capture that we really only expect this to be an STL container, we aren't using any API specific to an unordered map

Reply:
Do you mean here `std::unordered_map<std::string, InterfaceDescription> & state_interfaces_map` ? 

I'm not sure what traits can be used here. As both methods use the same set of arguments. If you have some ideas, please let me know :)

Reply:
Let's go with as-is ;)